internal/fs/tarfs/tarfs.go

```diff
@@ -48,6 +48,13 @@ func New(path string) (*TarFS, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to resolve absolute path for %s: %w", path, err)
 	}
+
+	extension := filepath.Ext(pathAbs)
+
+	if extension != ".tar" {
```

Comment:
Are compressed formats supported here like `tgz`?

Reply:
No, I believe they will not be.
Is there a list of extensions we need to support?

Reply:
I can add them to the check and add corresponding tests for them too.

Reply:
The current tests work with the files `hello-world.tar` and `test.tar`. From my understanding, `.tar`s are the only ones supported as of now.

Reply:
In `tfs.indexEntries()`, maybe we can return a friendly error if `tr.Next()` fails?

Reply:
I think `tfs.indexEntries` would indeed be a better place for this check. We can check if `EOF` is reached before any entries are added to `tfs`. This would infer it is an empty file. While a non-tar file will give us an `io.ErrUnexpectedEOF` while reading it.

We can then give the user our custom exported error.